eng/packages/http-client-csharp-mgmt/emitter/test/resolve-arm-resources-validation.test.ts

```diff
@@ -0,0 +1,1276 @@
+ *
+ * KNOWN BUGS IN resolveArmResources API (as of @azure-tools/typespec-azure-resource-manager v0.62.0):
+ *
+ * 1. Multiple singleton resources with different @singleton keys are not properly distinguished.
+ *    When both Employee and CurrentEmployee are defined with different singleton keys,
+ *    the API returns all resources pointing to the same Employee type only.
+ *    See: "singleton resource - demonstrates bug with multiple singletons" test case
+ *    See also: resolve-arm-resources-bug.test.ts for dedicated tests
+ *
+ * 2. Singleton child resources return incorrect resourceInstancePath.
+ *    For @singleton("current") @parentResource(Bar) BarSettings,
+ *    the API returns the parent path /bars/{barName} instead of /bars/{barName}/settings/current.
+ *    Existing TypeSpec: generator/TestProjects/Local/Mgmt-TypeSpec/bar.tsp (BarSettingsResource)
+ *    See: "singleton child resource - demonstrates path bug" test case below
+ *
+ * 3. Duplicate resources are returned for the same model.
+ *    The API can return 2-4 resolved resources for the same TypeSpec model.
+ *    Existing TypeSpec: generator/TestProjects/Local/Mgmt-TypeSpec/bar.tsp (BarSettingsResource)
+ *    See: "singleton child resource - demonstrates duplicate resources bug" test case below
+ *
+ * 4. Parent information is often undefined even when @parentResource is specified.
+ *    Existing TypeSpec: generator/TestProjects/Local/Mgmt-TypeSpec/bar.tsp (BarSettingsResource)
+ *    See: "singleton child resource - demonstrates parent undefined bug" test case below
+ *
+ * 5. Resource type for child resources uses the parent's type instead of the full type path.
+ *    Existing TypeSpec: generator/TestProjects/Local/Mgmt-TypeSpec/bar.tsp (BarSettingsResource)
+ *    See: "singleton child resource - demonstrates resource type bug" test case below
+ *
+ * These bugs prevent using resolveArmResources as a direct replacement for the
+ * decorator-based resource detection logic in resource-detection.ts.
+ * Once these bugs are fixed in the TypeSpec ARM library, the replacement can be completed.
+ */
+
+import { beforeEach, describe, it } from "vitest";
+import { createEmitterTestHost, typeSpecCompile } from "./test-util.js";
+import { TestHost } from "@typespec/compiler/testing";
+import {
+  resolveArmResources,
+  getSingletonResourceKey,
+  ResolvedResource
+} from "@azure-tools/typespec-azure-resource-manager";
+import { strictEqual, ok } from "assert";
+
+describe("resolveArmResources API Validation", () => {
+  let runner: TestHost;
+  beforeEach(async () => {
+    runner = await createEmitterTestHost();
+  });
+
+  /**
+   * Helper to get resource type string from ResourceType object
+   */
+  function getResourceTypeString(resource: ResolvedResource): string {
+    return `${resource.resourceType.provider}/${resource.resourceType.types.join("/")}`;
+  }
+
+  /**
+   * Helper to determine resource scope from path
+   */
+  function getResourceScope(
```

Comment:
This is *simpler* than the code that it is trying to test, so it doesn't seem like a good oracle, although it will work in many simple cases